snapshots/devmapper/snapshotter.go

```diff
@@ -376,7 +376,17 @@ func (s *Snapshotter) createSnapshot(ctx context.Context, kind snapshots.Kind, k
 		snapDeviceName := s.getDeviceName(snap.ID)
 		log.G(ctx).Debugf("creating snapshot device '%s' from '%s'", snapDeviceName, parentDeviceName)
 
-		err := s.pool.CreateSnapshotDevice(ctx, parentDeviceName, snapDeviceName, s.config.BaseImageSizeBytes)
+		// check if pool is running out of space and warn
```

Comment:
I think we should return an explicit error here rather than logging.

Reply:
If we error out, then the current operation will fail. Being below water mark would mean we are below reserve line but still not out of space.  Expectation is that user space will handle device mapper's event better e.g. dmeventd. This is a courtesy warning from containerd.

Reply:
I agree with @mxpv. People wouldn't notice warnings.

Reply:
If not sure if it worth it fail early. like let it consume the remaining space below water mark and then it'll fail anyway?
I'd rather rely on the event from DM to handle it correctly in advance TBH.

Reply:
Users are free to use lvm.conf to configure the thin pool accordingly. For example, users can specify thin_pool_autoextend_threshold and/or thin_pool_autoextend_percent to auto-extend the thin pool.

https://man7.org/linux/man-pages/man7/lvmthin.7.html#Thin_Topics

That being said, still I'd like to have a fail-fast behavior here by checking the low water mark. The error would be more clear than an error from mkfs.ext4.

Reply:
Yes, LVM has all the mechanisms to catch kernel uevent and let admin configure action. 

Docker fails to create snapshot as well when reaching thin pool low watermark.
https://github.com/docker/docker-ce/blob/master/components/engine/daemon/graphdriver/devmapper/deviceset.go#L861  

Looks like we should be good to return error instead of a warning.

Reply:
> still I'd like to have a fail-fast behavior here by checking the low water mark. The error would be more clear than an error from mkfs.ext4.

I don't like the idea to fail on water mark because this is not what people expect when they setup the water mark during thin-pool setup (lets say to set 30% to be able to fix things in advance, and instead containerd fails at 30%).
If we think that `mkfs` error message would be unclear, lets log the pool status right before returning error?